File: CCC-school-work/comporg/pnc2/kbeykirc/ruby/primeregba.rb
# prime number calculation thingy, now in ruby 
#
# to run, type: ruby primeregba.rb QTY
# 	where QTY is the number of primes to calculate

require "benchmark" # for timing

# check arguments
if ARGV.size == 1
	qty = ARGV[0].to_i
	
	if qty < 0
		abort("Must provide positive quantity!\n")
	end
else
	abort("Must provide one argument: quantity!\n")
end

# begin timing
time = Benchmark.realtime do
	num_primes = 0 # no primes have been found yet
	num = 2 # start checking numbers at 2
	factor = 2 # start checking factors at 2
	prime = 1 # number is prime until found to be composite
	
	# keep checking numbers until we find enough primes
	while num_primes < qty
		# check each number up to its square root
		while factor*factor <= num		
			# if number has a factor besides itself and 1,
			# it is composite
			if num%factor == 0
				prime = 0
				break
			end
			# go to next factor
			factor += 1
		end
		# if prime, print the number
		if prime == 1
			$stderr.printf "%d ", num
			num_primes += 1
		# otherwise restore to default
		else
			prime = 1
		end
		# restore factor and go to next number
		factor = 2
		num += 1
	end
	# a nice space for formatting
	$stderr.printf "\n"
end

$stdout.printf "%8.4f\n", time # print timing data

File: CCC-school-work/comporg/pnc2/kbeykirc/py/primeregba.py
# prime number calculation thingy, now in python 
#
# to run, type: python primeregba.py QTY 
#   where QTY is the number of primes to display 

import sys
import time

# check arguments 
if len(sys.argv) == 2:
    
    qty = int(sys.argv[1]) # set quantity 

    if(qty < 0):
        sys.stderr.write("Argument must be positive!\n")
        sys.exit
else:
    sys.stderr.write("Must provide one argument: quantity!\n")
    sys.exit

num_primes = 0 # number of primes 
num = 2 # number to start checking at
factor = 2 # factor to start checking at 
prime = 1 # is a number prime or not?
 
start = time.time() # begin timing 

# loop to find primes
while num_primes < qty:
    # check up until square root of number for possible factors
    while factor*factor <= num:
        # if a factor is found, number isn't prime
        if num%factor == 0:
            prime = 0
            break
        factor += 1

    # if prime, print it 
    if prime:
        sys.stderr.write("%d " % num) 
        num_primes += 1

    # restore defaults and go to next number
    prime = 1
    factor = 2
    num += 1

sys.stderr.write("\n") # a nice new line

end = time.time() # stop timing

total = end - start # calculate elapsed time

# print timing info 
sys.stdout.write("%8.4lf\n" % total)
